packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-customer-strategy.ts

```diff
@@ -0,0 +1,145 @@
+        private braintreeFastlaneUtils: BraintreeFastlaneUtils,
+    ) {}
+
+    async initialize({
+        methodId,
+        braintreeacceleratedcheckout,
+    }: CustomerInitializeOptions &
+        WithBraintreeFastlaneCustomerInitializeOptions): Promise<void> {
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" argument is not provided.',
+            );
+        }
+
+        const paymentMethod = await this.getValidPaymentMethodOrThrow(methodId);
+
+        this.isAcceleratedCheckoutEnabled =
+            !!paymentMethod.initializationData?.isAcceleratedCheckoutEnabled;
+
+        this.isFastlaneEnabled =
+            !!paymentMethod.initializationData?.isFastlaneEnabled;
+
+        if (!this.isFastlaneEnabled) {
+            if (this.isAcceleratedCheckoutEnabled) {
+                await this.braintreeFastlaneUtils.initializeBraintreeConnectOrThrow(
+                    paymentMethod.id,
+                    braintreeacceleratedcheckout?.styles,
+                );
+            }
+        } else {
+            await this.braintreeFastlaneUtils.initializeBraintreeFastlaneOrThrow(
+                paymentMethod.id,
+                braintreeacceleratedcheckout?.styles,
+            );
+        }
+
+        return Promise.resolve();
+    }
+
+    async deinitialize(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    async signIn(credentials: CustomerCredentials, options?: RequestOptions): Promise<void> {
+        await this.paymentIntegrationService.signInCustomer(credentials, options);
+    }
+
+    async signOut(options?: RequestOptions): Promise<void> {
+        await this.paymentIntegrationService.signOutCustomer(options);
+    }
+
+    async executePaymentMethodCheckout(
+        options?: ExecutePaymentMethodCheckoutOptions,
+    ): Promise<void> {
+        const { checkoutPaymentMethodExecuted, continueWithCheckoutCallback } = options || {};
+
+        if (typeof continueWithCheckoutCallback !== 'function') {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "continueWithCheckoutCallback" argument is not provided and it must be a function.',
+            );
+        }
+
+        if (this.isAcceleratedCheckoutEnabled || this.isFastlaneEnabled) {
+            const shouldRunAuthenticationFlow = await this.shouldRunAuthenticationFlow();
+
+            if (
+                checkoutPaymentMethodExecuted &&
+                typeof checkoutPaymentMethodExecuted === 'function'
+            ) {
+                checkoutPaymentMethodExecuted();
+            }
+
+            if (shouldRunAuthenticationFlow) {
+                if (!this.isFastlaneEnabled) {
+                    await this.braintreeFastlaneUtils.runPayPalConnectAuthenticationFlowOrThrow();
+                } else {
+                    await this.braintreeFastlaneUtils.runPayPalFastlaneAuthenticationFlowOrThrow();
+                }
+            }
```

Comment:
it might be better to update this conditions to be on 1 level instead of 2

if shouldRunAuthenticationFlow && this.isFastlaneEnabled

if shouldRunAuthenticationFlow && !this.isFastlaneEnabled